x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/LifecyclePolicy.java

```diff
@@ -75,11 +71,7 @@
      *            {@link LifecyclePolicy}.
      */
     public LifecyclePolicy(LifecycleType type, String name, Map<String, Phase> phases) {
-        if (type == null) {
-            this.type = TimeseriesLifecycleType.INSTANCE;
-        } else {
-            this.type = type;
-        }
+        this.type = type;
```

Comment:
We should make sure the type is not null here

Reply:
I changed things so that this should not be the case, but your last comment about updating all the unit tests to use TimeseriesLifecycleType changes behavior around this, so this may not even be here soon